### ISSUE TITLE
Super-Admin permission issue with saved searchs

### ISSUE THREAD

**Issue by cosmedd** [glpi-project/glpi#12751]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.3

### Bug description

With GLPI 10.0.2 a user with super-admin profile could see all his saved searches.

When upgrading to version 10.0.3 and trying to view the saved searches GLPI responds with the message _"You don't have permission to perform this action."_

It has been verified that the Super-Admin profile has all available permissions.

It has been verified that changing the user's profile to another one with less privileges such as Technincian allows to see the saved searches again.

### Relevant log output

```shell
2022-09-23 22:58:36 [382@glpi]
User ID: 382 tried to access or perform an action on /ajax/savedsearch.php?action=display_mine&itemtype=Ticket with insufficient rights. Additional information: No additional information given
	Stack Trace:
		/var/www/glpi/src/Search.php:560 displayRightError()
		/var/www/glpi/src/SavedSearch.php:1255 prepareDatasForSearch()
		/var/www/glpi/src/SavedSearch.php:834 execute()
		/var/www/glpi/src/SavedSearch.php:873 getMine()
		/var/www/glpi/ajax/savedsearch.php:88 displayMine()
```

### Your GLPI setup information

<details><summary>Information about system installation and configuration</summary><pre>GLPI 10.0.3-git-aitire-52f2f569c2 ( => /var/www/glpi)
Installation mode: GIT
Current language:en_GB

</details><details><summary>Server</summary><pre> 
Operating system: Linux glpi 4.19.0-21-amd64 #1 SMP Debian 4.19.249-2 (2022-06-30) x86_64
PHP 8.1.10 fpm-fcgi (Core, FFI, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, bz2, calendar, cgi-fcgi, ctype, curl, date,
	dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, igbinary, imap, intl, json, ldap, libxml, mbstring, mysqli, mysqlnd,
	openssl, pcre, pdo_mysql, posix, readline, redis, session, shmop, soap, sockets, sodium, standard, sysvmsg, sysvsem, sysvshm,
	tokenizer, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib)
Setup: max_execution_time="900" memory_limit="512M" post_max_size="50M" safe_mode="" session.save_handler="files"
	upload_max_filesize="50M" 
Software: Apache ()
	Mozilla/5.0 (X11; Linux x86_64; rv:104.0) Gecko/20100101 Firefox/104.0
Server Software: MySQL Community Server (GPL)
	Server Version: 5.7.39
	Server SQL Mode: STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
	Parameters: glpi@localhost/glpi
	Host info: Localhost via UNIX socket
	
PHP version (8.1.10) is supported.
Sessions configuration is OK.
Allocated memory is sufficient.
mysqli extension is installed.
Following extensions are installed: dom, fileinfo, json, simplexml.
curl extension is installed.
gd extension is installed.
intl extension is installed.
libxml extension is installed.
zlib extension is installed.
The constant SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES is present.
Database engine version (5.7.39) is supported.
The log file has been created successfully.
Write access to /var/www/glpi/files/_cache has been validated.
Write access to /var/www/glpi/config has been validated.
Write access to /var/www/glpi/files/_cron has been validated.
Write access to /var/www/glpi/files has been validated.
Write access to /var/www/glpi/files/_dumps has been validated.
Write access to /var/www/glpi/files/_graphs has been validated.
Write access to /var/www/glpi/files/_lock has been validated.
Write access to /var/www/glpi/files/_pictures has been validated.
Write access to /var/www/glpi/files/_plugins has been validated.
Write access to /var/www/glpi/files/_rss has been validated.
Write access to /var/www/glpi/files/_sessions has been validated.
Write access to /var/www/glpi/files/_tmp has been validated.
Write access to /var/www/glpi/files/_uploads has been validated.
Web access to files directory is protected
Sessions configuration is secured.
exif extension is installed.
ldap extension is installed.
openssl extension is installed.
zip extension is installed.
bz2 extension is installed.
Zend OPcache extension is installed.
Following extensions are installed: ctype, iconv, mbstring, sodium.
Write access to /var/www/glpi/marketplace has been validated.
Access to timezone database (mysql) is not allowed.

</details><details><summary>GLPI constants</summary><pre> 
GLPI_ROOT: /var/www/glpi
GLPI_CONFIG_DIR: /var/www/glpi/config
GLPI_VAR_DIR: /var/www/glpi/files
GLPI_MARKETPLACE_DIR: /var/www/glpi/marketplace
GLPI_USE_CSRF_CHECK: 1
GLPI_CSRF_EXPIRES: 7200
GLPI_CSRF_MAX_TOKENS: 100
GLPI_USE_IDOR_CHECK: 1
GLPI_IDOR_EXPIRES: 7200
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: 
GLPI_SERVERSIDE_URL_ALLOWLIST: Array
GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org
GLPI_INSTALL_MODE: GIT
GLPI_NETWORK_MAIL: glpi@teclib.com
GLPI_NETWORK_SERVICES: https://services.glpi-network.com
GLPI_MARKETPLACE_ALLOW_OVERRIDE: 1
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: 1
GLPI_USER_AGENT_EXTRA_COMMENTS: 
GLPI_DISABLE_ONLY_FULL_GROUP_BY_SQL_MODE: 1
GLPI_AJAX_DASHBOARD: 1
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: 0
GLPI_CENTRAL_WARNINGS: 1
GLPI_DOC_DIR: /var/www/glpi/files
GLPI_CACHE_DIR: /var/www/glpi/files/_cache
GLPI_CRON_DIR: /var/www/glpi/files/_cron
GLPI_DUMP_DIR: /var/www/glpi/files/_dumps
GLPI_GRAPH_DIR: /var/www/glpi/files/_graphs
GLPI_LOCAL_I18N_DIR: /var/www/glpi/files/_locales
GLPI_LOCK_DIR: /var/www/glpi/files/_lock
GLPI_LOG_DIR: /var/www/glpi/files/_log
GLPI_PICTURE_DIR: /var/www/glpi/files/_pictures
GLPI_PLUGIN_DOC_DIR: /var/www/glpi/files/_plugins
GLPI_RSS_DIR: /var/www/glpi/files/_rss
GLPI_SESSION_DIR: /var/www/glpi/files/_sessions
GLPI_TMP_DIR: /var/www/glpi/files/_tmp
GLPI_UPLOAD_DIR: /var/www/glpi/files/_uploads
GLPI_INVENTORY_DIR: /var/www/glpi/files/_inventories
GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/
GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/
GLPI_I18N_DIR: /var/www/glpi/locales
GLPI_VERSION: 10.0.3
GLPI_SCHEMA_VERSION: 10.0.3@a130db99c7d9b131c2e2ea59fe0d6260fe93d831
GLPI_MARKETPLACE_PRERELEASES: 
GLPI_MIN_PHP: 7.4.0
GLPI_MAX_PHP: 8.3.0
GLPI_YEAR: 2022

</details><details><summary>Libraries</summary><pre> 
htmlawed/htmlawed version 1.2.9 in (/var/www/glpi/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.6.0 in (/var/www/glpi/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.8 in (/var/www/glpi/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.5.0 in (/var/www/glpi/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/var/www/glpi/vendor/michelf/php-markdown/Michelf)
true/punycode in (/var/www/glpi/vendor/true/punycode/src)
iamcal/lib_autolink in (/var/www/glpi/vendor/iamcal/lib_autolink)
sabre/dav in (/var/www/glpi/vendor/sabre/dav/lib/DAV)
sabre/http in (/var/www/glpi/vendor/sabre/http/lib)
sabre/uri in (/var/www/glpi/vendor/sabre/uri/lib)
sabre/vobject in (/var/www/glpi/vendor/sabre/vobject/lib)
laminas/laminas-i18n in (/var/www/glpi/vendor/laminas/laminas-i18n/src)
laminas/laminas-servicemanager in (/var/www/glpi/vendor/laminas/laminas-servicemanager/src)
monolog/monolog in (/var/www/glpi/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/var/www/glpi/vendor/sebastian/diff/src)
donatj/phpuseragentparser in (/var/www/glpi/vendor/donatj/phpuseragentparser/src/UserAgent)
elvanto/litemoji in (/var/www/glpi/vendor/elvanto/litemoji/src)
symfony/console in (/var/www/glpi/vendor/symfony/console)
scssphp/scssphp in (/var/www/glpi/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/var/www/glpi/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/var/www/glpi/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/var/www/glpi/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/var/www/glpi/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/var/www/glpi/vendor/ramsey/uuid/src)
psr/log in (/var/www/glpi/vendor/psr/log/Psr/Log)
psr/simple-cache in (/var/www/glpi/vendor/psr/simple-cache/src)
psr/cache in (/var/www/glpi/vendor/psr/cache/src)
league/csv in (/var/www/glpi/vendor/league/csv/src)
mexitek/phpcolors in (/var/www/glpi/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/var/www/glpi/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/var/www/glpi/vendor/guzzlehttp/psr7/src)
glpi-project/inventory_format in (/var/www/glpi/vendor/glpi-project/inventory_format/lib/php)
wapmorgan/unified-archive in (/var/www/glpi/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/var/www/glpi/vendor/paragonie/sodium_compat/src)
symfony/cache in (/var/www/glpi/vendor/symfony/cache)
html2text/html2text in (/var/www/glpi/vendor/html2text/html2text/src)
symfony/css-selector in (/var/www/glpi/vendor/symfony/css-selector)
symfony/dom-crawler in (/var/www/glpi/vendor/symfony/dom-crawler)
twig/twig in (/var/www/glpi/vendor/twig/twig/src)
twig/string-extra in (/var/www/glpi/vendor/twig/string-extra)
symfony/polyfill-ctype not found
symfony/polyfill-iconv not found
symfony/polyfill-mbstring not found
symfony/polyfill-php80 not found
symfony/polyfill-php81 not found
symfony/polyfill-php82 in (/var/www/glpi/vendor/symfony/polyfill-php82)
phpCas version 1.3.6 in (/usr/share/php/CAS/source)

</details><details><summary>Plugins list</summary><pre> 
	fields               Name: Additional fields              Version: 1.17.3     State: Enabled                                 
		Install Method: Marketplace
	branding             Name: Branding                       Version: 1.3.1      State: Enabled                                 
		Install Method: Marketplace
	connections          Name: Connections                    Version: 10.0.0     State: Enabled                                 
		Install Method: Marketplace
	formcreator          Name: Form Creator                   Version: 2.13.1     State: Enabled                                 
		Install Method: Marketplace
	gantt                Name: gantt                          Version: 1.0.1      State: Enabled                                 
		Install Method: Marketplace
	gappessentials       Name: Gapp Essentials                Version: 2.0.1      State: Enabled                                 
		Install Method: Marketplace
	glpiinventory        Name: GLPI Inventory                 Version: 1.0.4      State: Enabled                                 
		Install Method: Marketplace
	glpihome             Name: glpihome                       Version: 0.1        State: Enabled                                 
		Install Method: Manual
	oauthsso             Name: oauthsso                       Version: 1.5.2      State: Enabled                                 
		Install Method: Marketplace
	reports              Name: Reports                        Version: 2.0.0      State: Enabled                                 
		Install Method: Manual
	manufacturersimports Name: Suppliers imports              Version: 3.0.2      State: Enabled                                 
		Install Method: Marketplace

</pre></details>

### Anything else?

![Captura desde 2022-09-23 22-22-47](https://user-images.githubusercontent.com/44313383/192056433-9b25d5fb-5cb4-4a14-af7e-c3856f45286f.png)

**Comment by cconard96** [comment]
I am not able to recreate the issue.

The line that the log lists is throwing the error should only occur if the General config option for "Items Seen" is disabled and the search contains the "Items Seen" criteria. It shouldn't matter which profile if used since it is a global setting.

Can you try disabling your plugins and testing again to see if there is something from them causing the issue?

**Comment by cosmedd** [comment]
Hi!

You are right. For some unknown reason we had the "Items seen" option disabled.

Now we do not reproduce the permissions error in the Super-Admin profile but unlike version 10.0.2, in the new one the saved searches of all users are shown.

Is this the expected behaviour or is it a bug?

In our opinion it is impractical to display the saved searches of all users when clicking on the "Lists" button.

We have tried to deactivate all plugins and have not observed any change in this issue.

Regards.

**Comment by cconard96** [comment]
That was a bug. The fix was made after the 10.0.3 release and can be found here:

https://github.com/glpi-project/glpi/pull/12692

**Comment by cconard96** [comment]
Note that even with the fix, users with the General Setup permission (config) of Update may see all saved searches even private ones of other users.
